### PR TITLE
python3Packages.nested-lookup: init at 0.2.23

### PR DESCRIPTION
**Pull request by tboerger**
###### Description of changes

Init new python module as a dependency for the upcoming package ansible-later/ansible-doctor

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nested-lookup/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    six
+  ];
+
+  meta = with lib; {
+    description = "Python functions for working with deeply nested documents (lists and dicts)";
+    homepage = "https://github.com/russellballestrini/nested-lookup";
+    license = licenses.publicDomain;
+    maintainers = with maintainers; [
+      tboerger
+    ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nested-lookup/default.nix:28]:
```suggestion
    maintainers = with maintainers; [ tboerger ];
```

File: pkgs/development/python-modules/nested-lookup/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-e4kAyOcG5PjDMJ694imxCfSHsVuajK4nuZAmrRHktkw=";
+  };
+
+  propagatedBuildInputs = [
+    six
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nested-lookup/default.nix:21]: Missing importsCheck.

**Comment by tboerger** [comment]
@SuperSandro2000 I have applied your requested changes.

**Comment by SuperSandro2000** [comment]
@ofborg build python3Packages.nested-lookup

**Comment by SuperSandro2000** [comment]
Please apply the following patch to fix ofborg

```diff
diff --git a/pkgs/development/python-modules/nested-lookup/default.nix b/pkgs/development/python-modules/nested-lookup/default.nix                                                            index fb25c5e4d65..7e56158ff14 100644                                                                                                                                                         --- a/pkgs/development/python-modules/nested-lookup/default.nix+++ b/pkgs/development/python-modules/nested-lookup/default.nix
@@ -1,8 +1,7 @@
 { buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , lib
-
-# pythonPackages
+, pytestCheckHook
 , six
 }:

@@ -10,15 +9,22 @@ buildPythonPackage rec {
   pname = "nested-lookup";
   version = "0.2.23";

-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-e4kAyOcG5PjDMJ694imxCfSHsVuajK4nuZAmrRHktkw=";
+  src = fetchFromGitHub {
+    owner = "russellballestrini";
+    repo = "nested-lookup";
+    # https://github.com/russellballestrini/nested-lookup/issues/47
+    rev = "c1b0421479efa378545bc71efa3b72882e8fec17";
+    sha256 = "sha256-jgfYLSsFLQSsOH4NzbDPKFIG+tWWZ1zTWcZEaX2lthg=";
   };

   propagatedBuildInputs = [
     six
   ];

+  checkInputs = [
+    pytestCheckHook
+  ];
+
   pythonImportsCheck = [ "nested_lookup" ];

   meta = with lib; {
 ```

**Comment by tboerger** [comment]
Thanks

**Pull request merged**